src/rdo.rs

```diff
@@ -875,16 +876,49 @@ pub fn rdo_partition_decision(
           }
         }).collect::<Vec<_>>();
 
+        let cw_checkpoint = cw.checkpoint();
+        let w_pre_checkpoint = w_pre_cdef.checkpoint();
+        let w_post_checkpoint = w_post_cdef.checkpoint();
+
         child_modes.extend(
           partitions
             .iter().zip(pmv_idxs)
             .map(|(&offset, pmv_idx)| {
+              let mode_decision =
               rdo_mode_decision(seq, fi, fs, cw, subsize, &offset,
                 &pmvs[pmv_idx])
                 .part_modes[0]
-                .clone()
+                .clone();
+
+                let (mode_luma, mode_chroma) = (mode_decision.pred_mode_luma, mode_decision.pred_mode_chroma);
+                let bo = offset;
+
+                let cfl = mode_decision.pred_cfl_params;
+                {
+                  let ref_frames = mode_decision.ref_frames;
+                  let mvs = mode_decision.mvs;
+                  let skip = mode_decision.skip;
+                  let mut cdef_coded = cw.bc.cdef_coded;
+                  let (tx_size, tx_type) = (mode_decision.tx_size, mode_decision.tx_type);
+
+                  cw.bc.set_tx_size(bo, tx_size);
+
+                  let mut mv_stack = Vec::new();
+                  let is_compound = ref_frames[1] != NONE_FRAME;
+                  let mode_context = cw.find_mvrefs(bo, ref_frames, &mut mv_stack, subsize, false, fi, is_compound);
+
+                  cdef_coded = encode_block_a(seq, cw, if cdef_coded  {w_post_cdef} else {w_pre_cdef},
```

Comment:
@xiphmont , I am using these two pre/post entropy coder state for cdef, pls check whether I do it correctly here and below.